openlibrary/templates/books/edit/about.html

```diff
@@ -21,7 +21,8 @@
             $for s in work.get_subjects():
                 $ s = s.replace('"', '""')
                 $subjects.append('"' + s + '"' if ',' in s else s)
-            <input type="text" name="work--subjects" id="about-subjects" value="$', '.join(subjects)" tabindex="0"/>
+            <textarea name="work--subjects" id="about-subjects" cols="55" rows="1" wrap="soft"
+            onkeyup="this.style.height = 'auto'; this.style.height = (3 + this.scrollHeight)+'px';" tabindex="0">$', '.join(subjects)</textarea>
```

Comment:
```suggestion
            $ subject_str = ', '.join(subjects)
            $ rows = len(subject_str) // 95
            <textarea name="work--subjects" id="about-subjects"  rows="$rows" wrap="soft"
            tabindex="0">$subject_str</textarea>
```
Your code works, but there are two issues:
1. We're trying to move all inline Javascript into `.js` files.
2. The textarea doesn't resize until a patron interacts with it.  This may frustrate some patrons, as it is unexpected.

We can estimate the number of rows in the template, and set  the textarea's `rows` equal to that.  I've noticed that a row contains about 95 characters.  By storing the joined subjects in a variable, dividing the length of that string by 95, and setting the `rows` equal to the result, I'm seeing good results on desktop-sized screens.  The textareas will still be a bit small on mobile devices, but they are larger and it should be fairly obvious that the content is scrollable.

`cols` isn't needed here, as it is overridden by our CSS rules.

Reply:
Sounds good, I just have a question: If they want to edit the field to add more items, would it change the rows?